server/src/main/java/org/elasticsearch/http/HttpContent.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.http;
+
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.core.Nullable;
+
+import java.util.function.Consumer;
+
+/**
+ * A super-interface for different HTTP content implementations
+ */
+public sealed interface HttpContent permits HttpContent.Full, HttpContent.Stream {
```

Comment:
Nit: I wonder if we could have different name for this class since netty also has a HttpContent.

Reply:
How about HttpBody?